fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java

```diff
@@ -415,6 +417,134 @@ public void scenario9() throws IOException {
+
+        logger.debug("user18 can patch - SPARQL INSERTs (ACL append): {}", id);
+        assertEquals(HttpStatus.SC_NO_CONTENT, getStatus(requestPatch));
+
+        logger.debug("user18 still can't delete (ACL append): {}", id);
+        assertEquals(HttpStatus.SC_FORBIDDEN, getStatus(requestDelete));
+
+        requestPatch.setEntity(new StringEntity("DELETE { <> <" + title.getURI() + "> \"Test title\" . } WHERE {}"));
+
+        logger.debug("user18 can not patch - SPARQL DELETEs (ACL append): {}", id);
+        assertEquals(HttpStatus.SC_FORBIDDEN, getStatus(requestPatch));
+
+        requestPatch.setEntity(null);
+
+        logger.debug("user18 can patch (is authorized, but bad request) - Empty SPARQL (ACL append): {}", id);
+        assertEquals(HttpStatus.SC_BAD_REQUEST, getStatus(requestPatch));
+
+        requestPatch.setHeader("Content-type", null);
+
+        logger.debug("user18 can not patch - Non SPARQL (ACL append): {}", id);
+        assertEquals(HttpStatus.SC_FORBIDDEN, getStatus(requestPatch));
+
+    }
+
+    @Test
+    public void scenario18Test2() throws IOException, UnsupportedEncodingException {
+        final String testObj = ingestObj("/rest/read_append_resource");
+        final String acl = ingestAcl("fedoraAdmin", "/acls/18/acl.ttl", "/acls/18/read_append.ttl");
+
+        final String id = "/rest/read_append_resource/" + getRandomUniqueId();
+        ingestObj(id);
+
+        logger.debug("user18 can't read (no ACL): {}", id);
+        final HttpGet requestGet = getObjMethod(id);
+        setAuth(requestGet, "user18");
+        assertEquals(HttpStatus.SC_FORBIDDEN, getStatus(requestGet));
+
+        logger.debug("user18 can't append (no ACL): {}", id);
+        final HttpPatch requestPatch = patchObjMethod(id);
+        setAuth(requestPatch, "user18");
+        requestPatch.setHeader("Content-type", "application/sparql-update");
+        requestPatch.setEntity(new StringEntity(
+                "INSERT { <> <" + title.getURI() + "> \"some title\" . } WHERE {}"));
+        assertEquals(HttpStatus.SC_FORBIDDEN, getStatus(requestPatch));
+
+        logger.debug("user18 can't delete (no ACL): {}", id);
+        final HttpDelete requestDelete = deleteObjMethod(id);
+        setAuth(requestDelete, "user18");
+        assertEquals(HttpStatus.SC_FORBIDDEN, getStatus(requestDelete));
+
+        linkToAcl("/rest/read_append_resource", acl);
+
+        logger.debug("user18 can read (ACL read, append): {}", id);
+        assertEquals(HttpStatus.SC_OK, getStatus(requestGet));
+
+        logger.debug("user18 can append (ACL read, append): {}", id);
+        assertEquals(HttpStatus.SC_NO_CONTENT, getStatus(requestPatch));
+
+        logger.debug("user18 still can't delete (ACL read, append): {}", id);
+        assertEquals(HttpStatus.SC_FORBIDDEN, getStatus(requestDelete));
+    }
+
+    @Test
+    public void scenario18Test3() throws IOException, UnsupportedEncodingException {
```

Comment:
It would be helpful to either have a description of what this test does or a link to resource describing the scenario.